bittensor/commands/unstake.py

```diff
@@ -294,3 +298,74 @@ def _run(cli: "bittensor.cli", subtensor: "bittensor.subtensor"):
             wait_for_inclusion=True,
             prompt=False,
         )
+
+
+class UnstakeTransferCommand:
+    """
+    Executes the ``swap`` command to unstake TAO tokens from all hotkeys and transfer them to the user's new coldkey on the Bittensor network.
+
+    Usage: Users need to specify the wallet from which to transfer from and provide the new coldkey address
+
+    Args:
+        wallet (bittensor.wallet): Bittensor wallet object to make transfer from.
+        dest (str, ss58_address or ed25519): Destination public key address of the new wallet.
+
+    Example usage::
+        btcli wallet swap --wallet.name test_wallet --new_wallet_address test_wallet_address
+
+    Note:
+        This command is important for users who wish to withdraw their stakes from their wallets and transfer them to a new wallet.
+    """
+
+    @staticmethod
+    def run(cli: "bittensor.cli"):
+        try:
+            subtensor: "bittensor.subtensor" = bittensor.subtensor(
+                config=cli.config, log_verbose=False
+            )
+            UnstakeTransferCommand._run(cli, subtensor)
+        finally:
+            if "subtensor" in locals():
+                subtensor.close()
+                bittensor.logging.debug("Closing Subtensor Connection")
+
+    @staticmethod
+    def _run(cli: "bittensor.cli", subtensor: "bittensor.subtensor"):
+        wallet = bittensor.wallet(config=cli.config)
+
+        subtensor.unstake_all_and_transfer(
+            wallet=wallet,
+            new_coldkey=cli.config.new_wallet_address,
+            wait_for_inclusion=True,
+            prompt=not cli.config.no_prompt,
+        )
+
+    @staticmethod
+    def check_config(config: "bittensor.config"):
+        if not config.is_set("wallet.name") and not config.no_prompt:
+            wallet_name = Prompt.ask("Enter wallet name", default=defaults.wallet.name)
+            config.wallet.name = str(wallet_name)
+
+        # Get destination wallet address.
+        if not config.new_wallet_address and not config.no_prompt:
+            new_wallet_address = Prompt.ask(
+                "Enter coldkey public key: (ss58 or ed2519)"
+            )
```

Comment:
This should say something more akin to "Enter new coldkey public key", otherwise it might imply to the user to enter their old coldkey public key.